apps/concept-catalog/app/catalogs/[catalogId]/concepts/import-results/[resultId]/import-result-details-page-client.tsx

```diff
@@ -19,54 +23,84 @@ interface Props {
+    mutationFn: async ({
+      catalogId,
+      importResultId,
+    }: {
+      catalogId: string;
+      importResultId: string;
+    }) => {
+      await deleteImportResult(catalogId, importResultId);
+      await router.push(`/catalogs/${catalogId}/concepts/import-results`);
+      await setShowDeleteConfirm(false);
+    },
+    onError: (error) => {
+      setIsDeleting(false);
       window.alert(error);
-    }
-  };
+    },
+  });
+
   const qc = useQueryClient();
 
   const handleDeleteClick = async () => {
     setShowDeleteConfirm(true);
   };
 
-  const confirmMutation = useMutation({
-    mutationFn: async () => await confirmImport(catalogId, importResult.id),
-    onSuccess: () => {
-      qc.invalidateQueries({
-        queryKey: ["refresh-import-result", catalogId, importResult.id],
-      });
+  const saveConceptMutation = useMutation({
+    mutationFn: async (externalId: string) => {
+      await saveImportedConcept(catalogId, importResult.id, externalId);
+    },
+    onSuccess: async () => {
+      await refetch();
     },
   });
 
-  const handleConfirmClick = () => {
-    confirmMutation.mutate();
-  };
-
   const cancelMutation = useMutation({
-    mutationFn: async () => await cancelImport(catalogId, importResult.id),
-    onSuccess: () => {
-      qc.invalidateQueries({
-        queryKey: ["refresh-import-result", catalogId, importResult.id],
-      });
+    mutationFn: async () => {
+      console.log("Is cancelling", isCancelling);
+      await cancelImport(catalogId, importResult.id);
+    },
+    onMutate: async () => {
+      setIsCancelling(true);
+    },
+    onSuccess: async () => {
+      const refetched = await refetch();
+      console.log("Refetched Import status data", refetched.data.status);
+      if (
+        isCancelling &&
+        refetched?.data.status !== ImportResutStatus.CANCELLED
+      ) {
+        console.log("should send cancel request again");
+        await cancelMutation.mutateAsync();
+      } else {
+        qc.setQueryData(
+          ["refresh-import-result", catalogId, importResult?.id],
+          (old: ImportResult) => ({
+            ...old,
+            status: ImportResutStatus.CANCELLED,
+            conceptExtractions: old?.conceptExtractions.map((ce) => ({
+              ...ce,
+              conceptExtractionStatus: ConceptExtractionStatus.CANCELLED,
+            })),
+          }),
+        );
+        await setIsCancelling(false);
+      }
     },
   });
 
-  const handleCancelClick = () => {
-    cancelMutation.mutate();
+  const handleCancelClick = async () => {
+    setIsCancelling(true);
+    await cancelMutation.mutateAsync();
   };
 
-  const shouldRefetch = (fetchedData) =>
-    fetchedData.status === "IN_PROGRESS" || fetchedData.status === "SAVING";
-
-  const isPendingConfirmation = (fetchedData) =>
-    fetchedData.status === "PENDING_CONFIRMATION";
+  const shouldRefetch = (fetchedData) => fetchedData?.status === "IN_PROGRESS";
```

Comment:
```suggestion
  const shouldRefetch = (fetchedData) => fetchedData?.status === ImportResutStatus.IN_PROGRESS;
```

du kan også vurdere å bare sende status direkte til funksjonen